molgenis-data-mapper/src/main/java/org/molgenis/data/mapper/algorithmgenerator/service/impl/AlgorithmGeneratorServiceImpl.java

```diff
@@ -0,0 +1,156 @@
+
+import static org.molgenis.data.mapper.mapping.model.AttributeMapping.AlgorithmState.GENERATED_HIGH;
+import static org.molgenis.data.mapper.mapping.model.AttributeMapping.AlgorithmState.GENERATED_LOW;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import javax.measure.quantity.Quantity;
+import javax.measure.unit.Unit;
+
+import org.apache.commons.lang3.StringUtils;
+import org.molgenis.data.AttributeMetaData;
+import org.molgenis.data.DataService;
+import org.molgenis.data.EntityMetaData;
+import org.molgenis.data.mapper.algorithmgenerator.bean.GeneratedAlgorithm;
+import org.molgenis.data.mapper.algorithmgenerator.generator.AlgorithmGenerator;
+import org.molgenis.data.mapper.algorithmgenerator.generator.NumericAlgorithmGenerator;
+import org.molgenis.data.mapper.algorithmgenerator.generator.OneToManyCategoryAlgorithmGenerator;
+import org.molgenis.data.mapper.algorithmgenerator.generator.OneToOneCategoryAlgorithmGenerator;
+import org.molgenis.data.mapper.algorithmgenerator.service.AlgorithmGeneratorService;
+import org.molgenis.data.mapper.mapping.model.AttributeMapping.AlgorithmState;
+import org.molgenis.data.mapper.service.UnitResolver;
+import org.molgenis.data.mapper.service.impl.AlgorithmTemplate;
+import org.molgenis.data.mapper.service.impl.AlgorithmTemplateService;
+import org.molgenis.data.semanticsearch.explain.bean.ExplainedAttributeMetaData;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static java.util.Objects.requireNonNull;
+
+import utils.AlgorithmGeneratorHelper;
+import utils.MagmaUnitConverter;
+
+public class AlgorithmGeneratorServiceImpl implements AlgorithmGeneratorService
+{
+	private final List<AlgorithmGenerator> generators;
+	private final AlgorithmTemplateService algorithmTemplateService;
+	private final UnitResolver unitResolver;
+	private final MagmaUnitConverter magmaUnitConverter = new MagmaUnitConverter();
+
+	@Autowired
+	public AlgorithmGeneratorServiceImpl(DataService dataService, UnitResolver unitResolver,
+			AlgorithmTemplateService algorithmTemplateService)
+	{
+		this.algorithmTemplateService = requireNonNull(algorithmTemplateService);
+		this.unitResolver = requireNonNull(unitResolver);
+		this.generators = Arrays.asList(new OneToOneCategoryAlgorithmGenerator(dataService),
+				new OneToManyCategoryAlgorithmGenerator(dataService), new NumericAlgorithmGenerator(unitResolver));
+	}
+
+	@Override
+	public String generate(AttributeMetaData targetAttribute, List<AttributeMetaData> sourceAttributes,
+			EntityMetaData targetEntityMetaData, EntityMetaData sourceEntityMetaData)
+	{
+		if (sourceAttributes.size() > 0)
+		{
+			for (AlgorithmGenerator generator : generators)
+			{
+				if (generator.isSuitable(targetAttribute, sourceAttributes))
+				{
+					return generator.generate(targetAttribute, sourceAttributes, targetEntityMetaData,
```

Comment:
No, at the moment it`ll not be possible. The generators are complementary to each other. Though it`d be good to have a way to keep track of to what situations those generators will apply, I don`t know how to do it yet